lib/adapters/parsers/columns.js

```diff
@@ -0,0 +1,129 @@
+'use strict';
+
+let base = require('./base');
+let Promise = require('bluebird');
+let readline = require('readline');
+
+// Simple parser for column-formatted text, such as when parsing the output of
+// unix commands like ps, lsof, etc.
+//
+// Each line is divided by whitespace. The first line is used to determine
+// where the columns start, and for each subsequent line, the values are
+// mapped into the best fit for the given offset.
+//
+// This is a bit tricky since some formats end up aligning the columns by the
+// left edge, some by the right, and sometimes the values hang over the
+// start/end offsets of the header, so we check for all cases as part of the
+// parsing.
+class ColumnParser extends base {
+    // Parse the given line into a set of columns based on whitespace. Returns
+    // an array of objects containing start / end offsets of the column  and the
+    // text value.
+    findColumns(line) {
+        let re = /\S+/gm;
+        let columns = [];
+        let match;
+        while ((match = re.exec(line))) {
+            columns.push({start: match.index, end: re.lastIndex, value: match[0]});
+        }
+        return columns;
+    }
+
+    // Given a set of headers and values parsed into column specs, return an
+    // object containing the best match based on either the start or the end
+    // offsets.
+    getValues(headers, columns, line) {
+        let hidx = 0, cidx = 0;
+
+        var result = {};
+
+        while (hidx < headers.length) {
+            let h = headers[hidx];
+            let c = columns[cidx];
+
+            // Fill in nulls when there are no more columns
+            if (cidx === columns.length) {
+                result[h.value] = null;
+                hidx++;
+
+            // For the last column, grab the full text from the original line to
+            // handle cases like ps output where there can be spaces in the
+            // final column
+            } else if (hidx === headers.length - 1) {
+                result[h.value] = line.substr(c.start);
+                break;
```

Comment:
By my calculations you can put `hidx++` after all the `if/else`s here and get rid of the three `hidx++`s and this `break.